File: jobtracker_back/src/main/java/com/arikaran/jobtracker/service/JobService.java
package com.arikaran.jobtracker.service;

import com.arikaran.jobtracker.entity.Job;
import com.arikaran.jobtracker.repository.JobRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class JobService {

    private final JobRepository jobRepository;

    public JobService(JobRepository jobRepository) {
        this.jobRepository = jobRepository;
    }

    // CREATE
    public Job addJob(Job job) {
        return jobRepository.save(job);
    }

    // READ ALL
    public List<Job> getAllJobs() {
        return jobRepository.findAll();
    }

    // READ ONE
    public Job getJobById(Long id) {
        return jobRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Job not found with id: " + id));
    }

    // UPDATE
    public Job updateJob(Long id, Job job) {

        Job existingJob = getJobById(id);

        existingJob.setCompanyName(job.getCompanyName());
        existingJob.setLocation(job.getLocation());
        existingJob.setJobRole(job.getJobRole());
        existingJob.setStatus(job.getStatus());
        existingJob.setAppliedDate(job.getAppliedDate());
        existingJob.setSalaryRange(job.getSalaryRange());
        existingJob.setJobType(job.getJobType());
        existingJob.setNotes(job.getNotes());
        existingJob.setJobPostingUrl(job.getJobPostingUrl());
        existingJob.setRecruiterUrl(job.getRecruiterUrl());

        return jobRepository.save(existingJob);
    }

    // DELETE
    public void deleteJob(Long id) {
        Job existingJob = getJobById(id);
        jobRepository.delete(existingJob);
    }
}

File: jobtracker_front/src/services/JobService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/jobs";

const JobService = {

  // GET all jobs
  getJobs: async () => {
    const response = await axios.get(API_URL);
    return response.data;
  },

  // GET one job
  getJob: async (id) => {
    const response = await axios.get(`${API_URL}/${id}`);
    return response.data;
  },

  // CREATE job
  addJob: async (job) => {
    const response = await axios.post(API_URL, job);
    return response.data;
  },

  // UPDATE job
  updateJob: async (id, job) => {
    const response = await axios.put(`${API_URL}/${id}`, job);
    return response.data;
  },

  // DELETE job
  deleteJob: async (id) => {
    await axios.delete(`${API_URL}/${id}`);
  },
};

export default JobService;